pymc4/_template_contexts.py

```diff
@@ -58,10 +60,13 @@ class InferenceContext(BaseContext):
 
     def __init__(self, tensors, expected_vars):
         self.vars = []
+        self._names = [var.name for var in expected_vars]
         self._tensors = {var.name: tensor for var, tensor in zip(expected_vars, tensors)}
```

Comment:
@twiecki It seems like we're going to have to be extra careful about passing in tensors and expected_vars in the correct order from wherever this is going to be called.

Is this a good design pattern? I honestly don't know, am just asking

Reply:
Yeah, the order has to keep constant in-between function calls. It's not ideal but also not totally crazy. I view this as a temporary work-around (i.e. hack) to get things working with `function()` which messes with the AST in its own way. There are different, more complex solutions to this, but they'd be harder to implement.

Reply:
Just briefly what are those solutions?

Reply:
The problem is that our current method changes the byte code of the function, not the source. That's totally fine if you just execute it, but if you then do AST parsing on that again, like auto-graph does, it will still see the old source code (because we didn't touch that). Auto-graph does actually change the source code of the function (so that's possible) but it's more complex and requires writing a file out to disk. The machinery they're building could be used for our purposes as well, but this short-cut is just way simpler.